src/CommonProviderImplementation/Helpers.fs

```diff
@@ -202,169 +233,210 @@ module internal ProviderHelpers =
     /// <param name="encodingStr"></param>
     /// <param name="fullTypeName"></param>
     /// <param name="maxNumberOfRows"></param>
-    let private parseTextAtDesignTime valueToBeParsedOrItsUri parseFunc formatName (tp:DisposableTypeProviderForNamespaces) 
-                                      (cfg:TypeProviderConfig) encodingStr resolutionFolder resource fullTypeName maxNumberOfRows =
-    
-        using (logTime "LoadingTextToBeParsed" valueToBeParsedOrItsUri) <| fun _ ->
-    
-        let tryGetResource() = 
-            if resource = "" then None else readResource(tp, resource)
-
-        let tryGetUri str =
-            match Uri.TryCreate(str, UriKind.RelativeOrAbsolute) with
-            | false, _ -> None
-            | true, uri ->
-                if str.Trim() = "" || not uri.IsAbsoluteUri && Seq.exists invalidChars.Contains str
-                then None else Some uri
-    
-        match tryGetResource() with
-        | Some res -> { Spec = parseFunc "" res
-                        IsUri = false
-                        IsResource = true }
-        | _ -> 
-
-        match tryGetUri valueToBeParsedOrItsUri with
-        | None -> 
-    
-            try
-                { Spec = parseFunc "" valueToBeParsedOrItsUri
+    let private parseTextAtDesignTime
+        valueToBeParsedOrItsUri
+        parseFunc
+        formatName
+        (tp: DisposableTypeProviderForNamespaces)
+        (cfg: TypeProviderConfig)
+        encodingStr
+        resolutionFolder
+        resource
+        fullTypeName
+        maxNumberOfRows
+        =
+
+        using (logTime "LoadingTextToBeParsed" valueToBeParsedOrItsUri)
```

Comment:
I will pre-change all explicit `using` to `use _holder = ...` to prevent this indentation happening.  Then we reformat